clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -408,11 +463,13 @@ private Optional<MetadataResponse.PartitionMetadata> updateLatestMetadata(
                 log.info("Resetting the last seen epoch of partition {} to {} since the associated topicId changed from {} to {}",
                         tp, newEpoch, oldTopicId, topicId);
                 lastSeenLeaderEpochs.put(tp, newEpoch);
+                this.equivalentResponseCount = 0;
                 return Optional.of(partitionMetadata);
             } else if (newEpoch >= currentEpoch) {
                 // If the received leader epoch is at least the same as the previous one, update the metadata
                 log.debug("Updating last seen epoch for partition {} from {} to epoch {} from new metadata", tp, currentEpoch, newEpoch);
                 lastSeenLeaderEpochs.put(tp, newEpoch);
+                this.equivalentResponseCount = 0;
```

Comment:
More of a question than a comment, but why do we reset `equivalentResponseCount` even if `newEpoch == currentEpoch` ? If epoch is available, then shouldn't we reset it only when `newEpoch > currentEpoch`?